database/test/db-admin-persona.test.js

```diff
@@ -1,55 +1,83 @@
+      const resultTenantname = await adminPersonas.updatePersona('test-tenant1 Non-existent', 'http://idp.dmgt.com.mock.localhost:3002/', 'auth0|test-persona2-REPLACE', 'should_not_replace_sub_claim', '', '')
       assert.strictEqual(resultTenantname, 'Update failed: A persona (sub_claim = auth0|test-persona2-REPLACE; user_context = ) does not exist on tenantname test-tenant1 Non-existent iss_claim http://idp.dmgt.com.mock.localhost:3002/')
     })
     it('Should fail to update for a nonexistent persona', async () => {
-      const resultTenantname = await postPersonas.updatePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'does-not-exist@dmgtoocto.com', 'should_not_replace_sub_claim', '', '')
+      const resultTenantname = await adminPersonas.updatePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'does-not-exist@dmgtoocto.com', 'should_not_replace_sub_claim', '', '')
       assert.strictEqual(resultTenantname, 'Update failed: A persona (sub_claim = does-not-exist@dmgtoocto.com; user_context = ) does not exist on tenantname test-tenant1 iss_claim http://idp.dmgt.com.mock.localhost:3002/')
     })
   })
 
   describe('Test DELETE personas', () => {
     it('Should fail to delete a persona not linked to a tenant', async () => {
-      const resultDelete = await postPersonas.deletePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'no-persona@dmgtoocto.com', '')
+      const resultDelete = await adminPersonas.deletePersona('test-tenant1', 'http://idp.dmgt.com.mock.localhost:3002/', 'no-persona@dmgtoocto.com', '')
       assert.strictEqual(resultDelete, 'Delete failed: A persona (sub_claim = no-persona@dmgtoocto.com; user_context = ) does not exist on tenantname test-tenant1 iss_claim http://idp.dmgt.com.mock.localhost:3002/')
     })
   })
+
+  describe('Test GET personas', () => {
+    const invalidPersonaKey = 0;
+    it('Should return a valid persona', async () => {
+      const persona = await adminPersonas.getPersona(1)
+      assert.strictEqual(persona.key, 1)
+    })
+    it('Should return undefined for invalid persona key', async () => {
+      const persona = await adminPersonas.getPersona(invalidPersonaKey)
+      assert.strictEqual(persona, undefined)
+    })
+  })
+
+  describe('Test GET personas permissions', () => {
+    const invalidPersonaKey = 0;
+    it('Should return an array of permissions for the persona', async function () {
+      const { personakey } = await usherDb('personapermissions').select('*').first() || {}
+      if (!personakey) {
+        this.skip()
+      }
+      const personaPermissions = await adminPersonas.getPersonaPermissions(personakey)
+      assert.equal(!!personaPermissions.length, true)
+    })
+    it('Should return an empty array', async () => {
```

Comment:
@k1ch  so the db function would return an empty list. Would the respective API still return an error code such as 404?

Reply:
Yes DB function returns an empty array since there is no record for invalid persona key. However, the API rejects the request by 404 since an invalid persona key is provided.